walkthroughs/5-adding-data-sync-graphql/walkthrough.adoc

```diff
@@ -452,13 +343,64 @@ Conflicting description from laptop
 
 . Bring both of your devices back online, a popup window should appear warning you about conflicts.
 
-
 [type=verification]
 ****
 Did the tasks sync as expected?
 ****
 
 [type=verificationFail]
 ****
-{standard-fail-text}
+Verify that you followed each step in the procedure above.  If you are still having issues, contact your administrator.
 ****
+
+. Close terminal window running server application
+
+[time=15]
+== Add authentication and authorization to the Data Sync application using Red Hat SSO
+
+In this task, we will configure both the frontend and the backend of the
+{data-sync-starter} with the {customer-sso-name}.
+
+DataSync starter has authentication and autorization enabled out of the box.
+Developers need to configure server and client application to use their keycloak instance
+and add required authorization rules to their model.
+
+=== Add authorization rule for Task deletion
+
+. Go to your GraphQL Schema `./server/src/schema/schema.qraphql`.
+Schema contains mutations section that is responsible for data modifications
+. In mutation section find `deleteTask(input: TaskInput): Task!`
+. Add GraphQL Directive on top of it `@hasRole(role: "admin")`
+This will only allow deletion for admin users.
+Roles can be also applied in generation process by utilizing graphback plugin
+. This directive is already defined in {data-sync-starter} and can be also applied
+to any new mutation or query created by users.
+We going to verify this directive in next steps
+
+=== Configuring Authentication for Keycloak (SSO) (Local setup)
```

Comment:
I'm not sure if we can do keycloak locally considered that there is one on RHMI. We need to copy paste entire instruction for keycloak from #39 that will be up to date

Reply:
Did you mean copy-paste it from #33? :)